### PR TITLE
feat: t4 - chapter2-l2-t4-files-no-extensions.sh

### PR DESCRIPTION
**Pull request by darliiin**
**Summary
Test Task**

Script should get a folder as an argument and return all files (including in the nested directories) but without the extension

**name the file**: `chapter2-l2-t4-files-no-extensions.sh`

**example of the call:**

./chapter2-l2-t4-files-no-extensions.sh ~/tmp/bazel

```
/tmp/bazel/tools/objc/dummy
/tmp/bazel/third_party/zlib/zutil
/tmp/bazel/third_party/zlib/uncompr
/tmp/bazel/third_party/zlib/trees
/tmp/bazel/third_party/zlib/test/minigzip
/tmp/bazel/third_party/zlib/test/infcover
/tmp/bazel/third_party/zlib/test/example
/tmp/bazel/third_party/zlib/inftrees

```

**my results**
`./chapter2-l2-t4-files-no-extensions.sh ~/Загрузки/a`
**output:**
`repos/iss-kubernetes-aws-deploy-key
repos/kustomization
repos/iss-kubernetes-aws-deploy-key
repos/iss-backend-deploy-key
repos/iss-frontend-deploy-key
repos/iss-backend-deploy-key
repos/iss-frontend-deploy-key
`

### CODE REVIEW

File: chapter2/tests/t4/chapter2-l2-t4-files-no-extensions.sh [hunk 1]
```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
```
Comment (kseniyashaydurova) [inline chapter2/tests/t4/chapter2-l2-t4-files-no-extensions.sh:1]: It still seems that something works incorrectly here
image

image

File: chapter2/tests/t4/chapter2-l2-t4-files-no-extensions.sh [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  exit 1
+fi
+
+# search folder contents, output content without extension
+find "$1" -type f | while read file; do
+
+  # check files that contain only one dot in the name
+  if [[ $(echo "$file" | grep -o "\." | wc -l) -eq 1 ]]; then
+
+    # if the file is hidden, then display the file name
```
Comment (kseniyashaydurova) [inline chapter2/tests/t4/chapter2-l2-t4-files-no-extensions.sh:17]:
```suggestion
    # if the file is dotfile
```

File: chapter2/tests/t4/chapter2-l2-t4-files-no-extensions.sh [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    if [[ "${file}" == *"/."* ]]; then
+      echo "${file}"
+    else
+      echo "${file%.*}"
+    fi
+
+  # check files that contain zero dots in the name
+  elif
+    [[ $(echo "$file" | grep -o "\." | wc -l) -eq 0 ]]; then
```
Comment (kseniyashaydurova) [inline chapter2/tests/t4/chapter2-l2-t4-files-no-extensions.sh:26]: Please move condition to the same line where `elif`, it is more common style of conditions, the same for below

**Review by kseniyashaydurova — CHANGES_REQUESTED** [review]

File: chapter2/tests/t4/chapter2-l2-t4-files-no-extensions.sh (hunk 3 above)
Comment (darliiin) [reply chapter2/tests/t4/chapter2-l2-t4-files-no-extensions.sh]: done

File: chapter2/tests/t4/chapter2-l2-t4-files-no-extensions.sh (hunk 2 above)
Comment (darliiin) [reply chapter2/tests/t4/chapter2-l2-t4-files-no-extensions.sh]: done

File: chapter2/tests/t4/chapter2-l2-t4-files-no-extensions.sh (hunk 1 above)
Comment (darliiin) [reply chapter2/tests/t4/chapter2-l2-t4-files-no-extensions.sh:1]: done

Comment (kseniyashaydurova) [reply chapter2/tests/t4/chapter2-l2-t4-files-no-extensions.sh:1]: Still doesn't work, you have the same problem as in 5th task
image

**Review by kseniyashaydurova — CHANGES_REQUESTED** [review]
:smiling_face_with_tear:

File: chapter2/tests/t4/chapter2-l2-t4-files-no-extensions.sh (hunk 1 above)
Comment (darliiin) [reply chapter2/tests/t4/chapter2-l2-t4-files-no-extensions.sh:1]: done

File: chapter2/tests/t4/chapter2-l2-t4-files-no-extensions.sh [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+      echo "$file"
+
+  # check files that contain more than one dot in the name
+  elif
+    [[ $(echo "$file" | grep -o "\." | wc -l) > 1 ]]; then
```
Comment (roman-mightydevops) [inline chapter2/tests/t4/chapter2-l2-t4-files-no-extensions.sh:31]:
```suggestion
    [[ $(echo "$file" | grep -o "\." | wc -l) -gt 1 ]]; then
```

**Review by roman-mightydevops — APPROVED** [review]

File: chapter2/tests/t4/chapter2-l2-t4-files-no-extensions.sh (hunk 4 above)
Comment (darliiin) [reply chapter2/tests/t4/chapter2-l2-t4-files-no-extensions.sh:31]: done

File: chapter2/tests/t4/chapter2-l2-t4-files-no-extensions.sh [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+      file_path=$(dirname "$file")
+      file_name="${file##*/}"
+
+      # remove the extension if the file name: contains more than 1 dot
+      # OR the file is not hidden and has the extension
+      if [[ ($(echo "$file_name" | grep -o "\." | wc -l) -ne 1) || \
```
Comment (kseniyashaydurova) [inline chapter2/tests/t4/chapter2-l2-t4-files-no-extensions.sh:37]: You do `$(echo "$file_name" | grep -o "\." | wc -l)` command lots of time, I think you can either move it to separate function or calculate it before your block with `if` conditions and save it to a separate variable (after the 12th line)

**Review by kseniyashaydurova — CHANGES_REQUESTED** [review]

File: chapter2/tests/t4/chapter2-l2-t4-files-no-extensions.sh (hunk 5 above)
Comment (darliiin) [reply chapter2/tests/t4/chapter2-l2-t4-files-no-extensions.sh]: done

**Review by kseniyashaydurova — APPROVED** [review]

**Pull request merged**